CMakeLists.txt

```diff
@@ -56,7 +56,7 @@ set(GINKGO_VERBOSE_LEVEL "1" CACHE STRING
 if(MSVC)
     set(GINKGO_COMPILER_FLAGS "" CACHE STRING
         "Set the required CXX compiler flags, mainly used for warnings. Current default is ``")
-elseif(GINKGO_BUILD_DPCPP OR CMAKE_CXX_COMPILER MATCHES "dpcpp")
```

Comment:
Maybe to prepare for more compilers in the future, we should add an option `GINKGO_BUILD_SYCL` and give `GINKGO_BUILD_DPCPP` the value of `GINKGO_BUILD_SYCL` by default.

Reply:
Maybe add them when we support other compilers not Intel's one?
Do we need to have `GINKGO_BUILD_DPCPP` and `GINKGO_BUILD_SYCL` at the same time?

Reply:
I think we could use `GINKGO_BUILD_DPCPP` to initialize `GINKGO_BUILD_SYCL` if it is available, otherwise only use `BUILD_SYCL`

Reply:
Yes, that was exactly my idea. This way we don't break build scripts at the moment, while allowing a switch to `BUILD_SYCL`.

Reply:
If I may suggest something, we could use a function like
```
gko_renamed_variable(GINKGO_BUILD_DPCPP GINKGO_BUILD_SYCL)
```
that expands to
```
if (EXISTS GINKGO_BUILD_DPCPP)
    if (EXISTS GINKGO_BUILD_SYCL)
        message(FATAL_ERROR "Both GINKGO_BUILD_SYCL and GINKGO_BUILD_DPCPP were specified, please remove the latter variable")
    else()
        message(WARNING "GINKGO_BUILD_DPCPP was deprecated, please use GINKGO_BUILD_SYCL instead")
        set(GINKGO_BUILD_SYCL ${GINKGO_BUILD_DPCPP})
    endif()
endif()
```
That would also allow us to implement things like #570 in the future

Reply:
I can try, but it is not really deprecated meaning.
after the first setting, setting DPCPP does not have the same functionality as previous one.

Reply:
done, I only provide the cache version because I think we do not need deprecation for internal variable.